lib/fog/libvirt/requests/compute/list_domains.rb

```diff
@@ -48,6 +48,23 @@ def boot_order xml
           xml_elements(xml, "domain/os/boot", "dev")
         end
 
+        # Foreman expects the firmware to be 'uefi_sb' if SB is enabled
+        def firmware(xml)
+          firmware_type = xml_elements(xml, "domain/os", "firmware").first || 'bios'
+          return 'uefi_sb' if firmware_type == 'efi' && secure_boot_enabled?(xml)
+
+          firmware_type
+        end
+
+        def secure_boot_enabled?(xml)
```

Comment:
Another thought I had: you can implement `secure_boot?` on the `Server` object that inspects the attributes.

Reply:
For what use?